src/EasyCaching.CSRedis/DefaultCSRedisCachingProvider.cs

```diff
@@ -55,14 +55,14 @@ public partial class DefaultCSRedisCachingProvider : EasyCachingAbstractProvider
            string name,
            IEnumerable<EasyCachingCSRedisClient> clients,
            IEasyCachingSerializer serializer,
-           IOptionsMonitor<RedisOptions> options,
+           RedisOptions options,
            ILoggerFactory loggerFactory = null)
         {
             this._name = name;
             this._serializer = serializer;
-            this._options = options.CurrentValue;
+            this._options = options;
             this._logger = loggerFactory?.CreateLogger<DefaultCSRedisCachingProvider>();
-            this._cache = clients.FirstOrDefault(x => x.Name.Equals(_name));
+            this._cache = clients.Single(x => x.Name.Equals(_name));
```

Comment:
Is `Single` better than `FirstOrDefault` here?

Reply:
If there are multiple providers with the same name, an error will be reported here.

Reply:
It sounds good!